internal/cli/service_build_image.go

```diff
@@ -0,0 +1,112 @@
+package cli
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	cliTypes "github.com/docker/cli/cli/config/types"
+	"github.com/rwx-cloud/cli/internal/api"
+)
+
+func (s Service) BuildImage(config BuildImageConfig) error {
+	if err := config.Validate(); err != nil {
+		return err
+	}
+
+	runResult, err := s.InitiateRun(InitiateRunConfig{
+		InitParameters: config.InitParameters,
+		RwxDirectory:   config.RwxDirectory,
+		MintFilePath:   config.MintFilePath,
+		NoCache:        config.NoCache,
+		TargetedTasks:  []string{config.TargetTaskKey},
+	})
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(s.Stdout, "Building image for %s\n", config.TargetTaskKey)
+	fmt.Fprintf(s.Stdout, "Run URL: %s\n\n", runResult.RunURL)
+	fmt.Fprintf(s.Stdout, "Polling for build completion...\n")
+
+	ctx, cancel := context.WithTimeout(context.Background(), config.Timeout)
+	defer cancel()
+
+	var taskID string
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timeout waiting for build to complete after %s\n\nThe build may still be running. Check the status at: %s", config.Timeout, runResult.RunURL)
+		default:
+		}
+
+		result, err := s.APIClient.TaskStatus(api.TaskStatusConfig{
+			RunID:   runResult.RunId,
+			TaskKey: config.TargetTaskKey,
+		})
+		if err != nil {
+			return fmt.Errorf("failed to get build status: %w", err)
+		}
+
+		switch result.Status {
+		case api.TaskStatusPending:
+			backoffMs := result.BackoffMs
+			if backoffMs == 0 {
+				backoffMs = 1
+			}
```

Comment:
what's the case where this is 0 (and why would we want to immediately retry in that case)?